src/runtime/lib/telemetry/index.ts

```diff
@@ -150,6 +218,35 @@ export class TelemetryClient {
             });
     }
 
+    /**
+     * check data and timestamp for updating telemetry data
+     * @returns 
+     */
+    private shouldUpdateData(): boolean {
+        if (ext.context.globalState.get(TELEMETRY_DATA) == undefined && 
+        (ext.context.globalState.get(TELEMETRY_DATA) as []).length == 0 ) {
+            return false;
+        }
+        if (ext.context.globalState.get(TELEMETRY_UPDATE_TIME) == undefined) {
+            const now = new Date().getTime();
+            ext.context.globalState.update(TELEMETRY_UPDATE_TIME, now);
+            return false;
+        }
+        const now = new Date().getTime();
+        const pre = ext.context.globalState.get(TELEMETRY_UPDATE_TIME) as number;
+        const diff = Math.abs(now - pre) / MINUTE;
+        if (diff < NUMBER_OF_INTERVALS) {
+            return false;
+        }
+        return true;
+    }
+
+    private async resetStoredStates(): Promise<void> {
+        this.data.actions = [];
+        await ext.context.globalState.update(TELEMETRY_DATA, undefined);
+        await ext.context.globalState.update(TELEMETRY_UPDATE_TIME, undefined);
```

Comment:
Rather than maintaining two different decoupled variables, can we have a single map containing both data and update time, and thus organizing/coupling the related info at one place?

Reply:
Updated to use a dictionary {time, actions[]}